### PR TITLE
feature/typescript config

### PR DESCRIPTION
**Pull request by DanC5**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

First draft of setting up typescript for the library. TS > JS conversion seems to be working properly, though the Prettier setup still isn't full happy on commit. Will jump back into this Monday, but wanted to open a PR in case anyone else wants to have a look at the config.

### CODE REVIEW

**Comment by DanC5** [comment]
@jaapbakker88 this is ready for review, those a node version mismatch looks to be failing the build process

**Comment by nhannah** [comment]
> @jaapbakker88 this is ready for review, those a node version mismatch looks to be failing the build process

FYI this is because Node V.11 has reached its end of life. Long term support for odd number releases is not done. There is info on this here: https://github.com/nodejs/Release/ .

**Comment by jaapbakker88** [comment]
@DanC5 @nhannah dropped support for node 11 and 13 and merged that into this branch. There's a lot of linting errors though.

**Comment by DanC5** [comment]
> @DanC5 @nhannah dropped support for node 11 and 13 and merged that into this branch. There's a lot of linting errors though.

Yeah adding in the TS linting recommendations threw a ton of warnings/errors. We'll need discuss how we want to handle these.

**Comment by vishalnarkhede** [comment]
@DanC5 @nhannah  `types/index.d.ts` - this is autogenerated file, right? And currently we are only exposing `types/stream-chat/index.d.ts` as source of types?

File: src/connection.ts [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+const isErrorEvent = (
+  res: WebSocket.CloseEvent | WebSocket.Data | WebSocket.ErrorEvent,
+): res is WebSocket.ErrorEvent => (res as WebSocket.ErrorEvent).message !== undefined;
+
+type ConnectionOpen = {
+  connection_id?: string;
+  [key: string]: unknown;
+};
+
+type Constructor = {
```
Comment (vishalnarkhede) [inline src/connection.ts:21]: Do we need this separate type for Constructor? Will it be used anywhere? @DanC5

**Pull request closed without merging**